gradio/chat_interface.py

```diff
@@ -270,11 +300,50 @@ def _setup_events(self):
                 api_name=False,
             )
 
-    def _setup_api(self):
-        if inspect.isgeneratorfunction(self.fn):
-            api_fn = self._api_stream_fn
-        else:
-            api_fn = self._api_submit_fn
+    def _setup_stop_events(
+        self, event_trigger: EventListenerMethod, event_to_cancel: Dependency
+    ) -> None:
+        if self.stop_btn and self.is_generator:
+            if self.submit_btn:
+                event_trigger(
+                    lambda: (Button.update(visible=False), Button.update(visible=True)),
+                    None,
+                    [self.submit_btn, self.stop_btn],
+                    api_name=False,
+                    queue=False,
+                )
+                event_to_cancel.then(
+                    lambda: (Button.update(visible=True), Button.update(visible=False)),
+                    None,
+                    [self.submit_btn, self.stop_btn],
+                    api_name=False,
+                    queue=False,
+                )
+            else:
+                event_trigger(
+                    lambda: Button.update(visible=True),
+                    None,
+                    [self.stop_btn],
+                    api_name=False,
+                    queue=False,
+                )
+                event_to_cancel.then(
```

Comment:
I don't think this is right. We're making `self.submit_btn` visible but in this branch submit_btn was not defined. I think we mean to hide the stop_btn in this case

Reply:
Nice catch, fixed